chain/chain/src/chain.rs

```diff
@@ -4771,17 +4770,13 @@ impl Chain {
         receipts: &[Receipt],
         shard_layout: &ShardLayout,
     ) -> Vec<CryptoHash> {
-        let shard_ids: Vec<_> = shard_layout.shard_ids().collect();
-        if shard_ids.len() == 1 {
```

Comment:
We do not need to special case handling of one shard as we do not have that few shards in production.